### ISSUE TITLE
[Bug]: Neovim gets stuck when opening a C# file

### ISSUE THREAD

**Issue by Nekketsu** [Bekaboo/dropbar.nvim#89]
### Description

When I open a C# file, Neovim gets stuck. It doesn't do anything, it doesn't react to keyboard or mouse inputs, etc.
I don't know if it is a Neovim problem, or a Dropbar problem, or a dotnet problem.

### nvim version

NVIM v0.10.0-dev-0aea8fa

### dropbar.nvim version

d9e2b240e9d4fe42fc74c9e40127f3217f00e14b

### Operating system and version

Windows 10

### Minimal config

```Lua
local tmp = vim.env.TMPDIR
  or vim.env.TEMPDIR
  or vim.env.TMP
  or vim.env.TEMP
  or '/tmp'
local data = tmp .. '/' .. (vim.env.NVIM_APPNAME or 'nvim')
local packages_root = data .. '/site'
local cloned_root = packages_root .. '/pack/packages/start'
local cloned_path = cloned_root .. '/dropbar.nvim'
local url = 'https://github.com/Bekaboo/dropbar.nvim.git'

vim.fn.mkdir(cloned_root, 'p')
vim.opt.pp:prepend(packages_root)
vim.opt.rtp:prepend(packages_root)
vim.opt.mousemoveevent = true

if not vim.loop.fs_stat(cloned_path) then
  vim.fn.system({ 'git', 'clone', url, cloned_path })
end

require('dropbar').setup()
```

### Steps to reproduce

1. Save the configuration above as `minimal.lua`
2. `nvim --clean -u minimal.lua`
3. Open a C# file, e.g.: `:e user.cs`:
```cs
public class User
{
    public string UserName { get; private set; }
    public string Mode { get; private set; }
    public string Unused { get; private set; }
    public string RealName { get; private set; }

    public User(string user, string mode, string unused, string realName)
    {
        UserName = user;
        Mode = mode;
        Unused = unused;
        RealName = realName;
    }
}
```

### Expected behavior

Neovim should open the C# file and show the dropbar.

### Actual behavior

Neovim gets stuck. It shows the contents of the `.cs` file but it doesn't show the dropbar, and it gets stuck. It doesn't do anything, it doesn't react to keyboard or mouse inputs, etc. The only solution I have found is killing the Neovim application.

**Comment by willothy** [comment]
That repo is not public, but also a minimal init would be helpful as cloning an entire config and potentially overwriting plugin manager cache dirs is not ideal, and makes narrowing down the source of the issue quite difficult.

**Comment by Bekaboo** [comment]
@Nekketsu As @willothy suggests, I am waiting for a minimal config to debug.

**Comment by Nekketsu** [comment]
I have been able to leave only the minimal config. It seems that Neovim gets stuck with the default `dropbar.nvim` configuration.
If I open any `.lua` file and `dropbar.nvim` works perfectly. However, if I open any `.cs` file `dropbar.nvim` gets stuck. For example `User.cs`:
```cs
public class User
{
    public string UserName { get; private set; }
    public string Mode { get; private set; }
    public string Unused { get; private set; }
    public string RealName { get; private set; }

    public User(string user, string mode, string unused, string realName)
    {
        UserName = user;
        Mode = mode;
        Unused = unused;
        RealName = realName;
    }
}
```

**Comment by Nekketsu** [comment]
I have tried to rename a `.lua` file as `.cs` file, and Neovim doesn't get stuck. So it seems that it gets stuck depending on the contents of the file, not in the file extension.

**Comment by Bekaboo** [comment]
@Nekketsu Please do not use a plugin manager (e.g. lazy.nvim) in the minimal config.

**Comment by Nekketsu** [comment]
I have updated the initial post, so now it doesn't use a plugin manager. The repro steps are the same.

**Comment by Bekaboo** [comment]
@Nekketsu Cannot reproduce on Linux. Maybe it is related to the differences in the design of the filesystems on Linux and Windows, which I have no idea how to fix.


https://github.com/Bekaboo/dropbar.nvim/assets/76579810/4aea0132-ee24-440d-9265-a20f2a474c17

**Comment by Nekketsu** [comment]
Thank for looking at it.

Any idea how to debug it?
For me Neovim just gets stuck and I don't where to see any log, or how to progress with that.

I would like to see the call that is making everything to freeze, but I don't know how to.

**Comment by Bekaboo** [comment]
@Nekketsu You can remove the `path` source from sources used (keep only the lsp source) like this and see if the problem persists:

```lua
local sources = require('dropbar.sources')

require('dropbar').setup({
  bar = {
    sources = {
      sources.lsp,
    },
  },
})
```

**Comment by Nekketsu** [comment]
@Bekaboo , thank you very much. I have tried with `sources.lsp` and it is working (however it is not showing anything in the `.cs` file unless it is part of a C# project, which could make sense due to the C# LSP implementation).
I have tried also to add `sources.treesitter` and it is also working.

So, it seems that your theory is confirmed, and it might be a problem regarding to the difference in the design of the filesystems on Linux and Windows. Could it be related to the path separator slash '/' (Linux)  vs backslash `\` (Windows)?

Please, let me know if I can do something to help with this.

**Comment by Bekaboo** [comment]
There is a pull request to improve cross-platform compatibility on Windows: https://github.com/Bekaboo/dropbar.nvim/pull/42, since I do not use Windows myself, I am afraid there's very little thing I can do to fix the issue. If you are interested you can take a look at the implementation of the `path` source: https://github.com/Bekaboo/dropbar.nvim/blob/master/lua/dropbar/sources/path.lua. The implementation is simple and contains only 100+ lines.

**Comment by Bekaboo** [comment]
Should be fixed by #111

**Comment by Nekketsu** [comment]
Thank you, it works perfectly now in Windows.
